libs/model/src/policies/ContestWorker.policy.ts

```diff
@@ -39,17 +39,20 @@ export function ContestWorker(): Policy<typeof inputs> {
         // content url only contains path
         const contentUrl = new URL(fullContentUrl).pathname;
 
-        for (const contestManager of community!.contest_managers!) {
-          const { address: userAddress } = (await models.Address.findByPk(
-            payload!.address_id,
-          ))!;
-          await contestHelper.addContent(
-            chainNodeUrl!,
-            contestManager.contest_address!,
-            userAddress,
-            contentUrl,
-          );
-        }
+        // TODO: optimize with bulk fetch
+        await Promise.allSettled(
```

Comment:
what happens if one fails?

Reply:
I guess we should check if any failed, then throw at the end so that it can retry. As long as it's idempodent it's fine.

Reply:
so, it should be all or nothing right?

Reply:
This logic can never be "all or nothing"– some onchain calls may randomly fail at any time, and the ones that succeed cannot be reverted. The best we can do is use `allSettled` to reduce the surface area of failure and give each async call a fair chance, but still fail the handler if a single failure was detected so it retries.

Reply:
I mean all or nothing to retry the event, and the processing of the next retry handles the only the ones that failed -> idempotent

Reply:
Yep. So regarding "only the ones that failed"– does that mean we can query the projection from the worker policy to check?

Reply:
<img width="345" alt="Screenshot 2024-05-31 at 12 20 06 PM" src="https://github.com/hicommonwealth/commonwealth/assets/41707007/e608965a-3bba-443a-8161-d2d92e3bbfc5">
😆

Reply:
Ok, wasn't sure if there was potentially a convention *against* doing that.

So apparently policies can both query projections and *indirectly* write projections (which it's doing via the `addContent` call).

Reply:
"indirectly" is the key word... policies invoke commands that emit events that can be projected